oem/firstvoices/android/app/src/main/java/com/firstvoices/keyboards/FVShared.java

```diff
@@ -412,4 +427,51 @@ void upgradeTo12() {
 
         saveLoadedKeyboardList();
     }
+
+    /// Upgrades the keyboard list Keyman Engine for Android uses from an earlier version to 14.0
+    // because the FirstVoices keyboard packageIDs changed to "fv_all".
+    // Note: KMManager.initialize will eventually migrate KMFilename_KeyboardsList to a JSON file
+    @SuppressWarnings("unchecked")
+    void upgradeTo14() {
+        String TAG = "upgradeTo14";
+        File file = new File(context.getDir("userdata", Context.MODE_PRIVATE), KMManager.KMFilename_KeyboardsList);
+        if(!file.exists()) {
```

Comment:
I'm puzzled by this -- this looks it could never run, because `saveUpdateTo14List` is only ever called by `upgradeTo14`, so when does this ever run?

Reply:
Oh never mind me, I think I understand. Ths file will exist from a 12.0 version and then you just update and overwrite it before the final migration done by KMManager... which then deletes it. :)

Reply:
It's a subtle filename difference I discovered while debugging.

upgradeTo12() migrates and removes FVUpgrade_KeyboardList (l.35) - keyboard_list.dat.
upgradeTo14() uses the legacy Keyman Engine for Android file KMManager.KMFilename_KeyboardsList -  keyboard**s**_list.dat.

So upgradeTo14() will run when an FV user is updating their app to 14.0

Reply:
🤯 🤯 🤯